platform/frontend/src/app/chat/page.tsx

```diff
@@ -361,113 +337,122 @@ export default function ChatPage() {
         });
       }
     }
-
-    // Clear the edit flag when status changes to ready (streaming finished)
-    if (status === "ready" && userMessageJustEdited.current) {
-      userMessageJustEdited.current = false;
-    }
   }, [
     conversationId,
     conversation,
     setMessages,
     sendMessage,
     status,
-    messages.length,
+    messages,
   ]);
 
-  // Merge database UUIDs from backend into local message state
-  // This runs after streaming completes and backend query has fetched
-  useEffect(() => {
-    if (
-      !setMessages ||
-      !conversation?.messages ||
-      conversation.id !== conversationId ||
-      status === "streaming" ||
-      status === "submitted"
-    ) {
-      return;
-    }
+  const focusTextarea = useCallback(() => {
+    requestAnimationFrame(() => {
+      textareaRef.current?.focus();
+    });
+  }, []);
 
-    // Only merge IDs if backend has same or more messages than local state
-    if (conversation.messages.length < messages.length) {
-      return;
-    }
+  const handleSubmit = useCallback(
+    (
+      // biome-ignore lint/suspicious/noExplicitAny: AI SDK PromptInput files type is dynamic
+      message: { text?: string; files?: any[] },
+      e: FormEvent<HTMLFormElement>,
+    ) => {
+      e.preventDefault();
+      if (!sendMessage) return;
+
+      const trimmed = message.text?.trim();
+      if (!trimmed) return;
+
+      // If a message is currently being generated, queue this message instead
+      if (status === "submitted" || status === "streaming") {
+        chatSession?.addQueuedMessage?.({
+          id:
+            typeof crypto !== "undefined" && "randomUUID" in crypto
+              ? crypto.randomUUID()
+              : `queued-${Date.now()}-${Math.random().toString(36).slice(2, 9)}`,
```

Comment:
why the conditional check for `crypto`, wouldn't this either be always available or not?